RecoParticleFlow/PFProducer/src/PFAlgo.cc

```diff
@@ -1076,30 +1047,29 @@ void PFAlgo::elementLoop(const reco::PFBlock& block,
     std::multimap<double, unsigned> gsfElems;
     block.associatedElements(iTrack, linkData, gsfElems, reco::PFBlockElement::GSF);
 
-    if (hcalElems.empty() && debug_) {
-      cout << "no hcal element connected to track " << iTrack << endl;
+    if (hcalElems.empty()) {
+      LogDebug("PFAlgo::elementLoop") << "no hcal element connected to track " << iTrack;
     }
 
     // will now loop on associated elements ...
     // typedef std::multimap<double, unsigned>::iterator IE;
 
     bool hcalFound = false;
 
-    if (debug_)
-      cout << "now looping on elements associated to the track" << endl;
+    LogDebug("PFAlgo::elementLoop") << "now looping on elements associated to the track: ecalElems";
 
     // ... first on associated ECAL elements
     // Check if there is still a free ECAL for this track
     for (auto const& ecal : ecalElems) {
       unsigned index = ecal.second;
       // Sanity checks and optional printout
       PFBlockElement::Type type = elements[index].type();
-      if (debug_) {
-        double dist = ecal.first;
-        cout << "\telement " << elements[index] << " linked with distance = " << dist << endl;
-        if (!active[index])
-          cout << "This ECAL is already used - skip it" << endl;
-      }
+#ifdef EDM_ML_LOGDEBUG
```

Comment:
What is the scope of these directives? Since you are going to use them, why not to let the previous ```cout```'s, once protected by them?
Why there are both  ```#ifdef EDM_ML_LOGDEBUG``` and ```#ifdef EDM_ML_DEBUG```? Is it a typo?

Reply:
my bad, should be `EDM_ML_DEBUG`